### PR TITLE
CLI: print all available dynamic config keys

### PR DESCRIPTION
**Pull request by mantas-sidlauskas**
**What changed?**
Adding CLI command to print out all available Dynamic Config keys with all details: name, description and default value.
For usability, `--format` flag can be used to print the list in preferred format (json, table, go template)

**Why?**
Currently, there are 369 configuration keys. 
This command will help to find needed key and also check what is the default value.

Snippet of the output:
```
$ cadence admin config describe
+-----------------------------------------------------------------------+---------------------------------------------------------------+--------------------------------+
|                                 NAME                                  |                          DESCRIPTION                          |         DEFAULT VALUE          |
+-----------------------------------------------------------------------+---------------------------------------------------------------+--------------------------------+
| admin.errorInjectionRate                                              | dminErrorInjectionRate is                                     |                              0 |
|                                                                       | the rate for injecting random                                 |                                |
|                                                                       | error in admin client                                         |                                |
| frontend.decisionResultCountLimit                                     | FrontendDecisionResultCountLimit                              |                              0 |
|                                                                       | is max number of decisions per                                |                                |
|                                                                       | RespondDecisionTaskCompleted                                  |                                |
|                                                                       | request                                                       |                                |
| frontend.disableListVisibilityByFilter                                | DisableListVisibilityByFilter                                 |                          false |
|                                                                       | is config to disable list                                     |                                |
|                                                                       | open/close workflow using                                     |                                |
|                                                                       | filter                                                        |                                |
| frontend.domainFailoverRefreshInterval                                | DomainFailoverRefreshInterval                                 |                            10s |
|                                                                       | is the domain failover refresh                                |                                |
|                                                                       | timer                                                         |                                |
| frontend.domainFailoverRefreshTimerJitterCoefficient                  | DomainFailoverRefreshTimerJitterCoefficient                   |                            0.1 |
|                                                                       | is the jitter for domain failover refresh                     |                                |

```

### CODE REVIEW

File: tools/cli/admin.go [hunk 1]
```diff
@@ -1280,5 +1280,14 @@ func newAdminConfigStoreCommands() []cli.Command {
 				AdminListDynamicConfig(c)
 			},
 		},
+		{
+			Name:    "describe",
```
Comment (Shaddoll) [inline tools/cli/admin.go:1284]: `list` seems to be a better name

Comment (mantas-sidlauskas) [reply tools/cli/admin.go:1284]: Yeah, naming is hard.

There is `list-dynamic-config` argument already, and I'm proposing to rename it to `list` [in this PR](https://github.com/uber/cadence/pull/5089/files#diff-44b0fcadee023fdd151d83020b84d464e980b288f49d165b1115f2dd43c075eaR1280)

**Review by Shaddoll — APPROVED** [review]

File: tools/cli/admin.go (hunk 1 above)
Comment (Shaddoll) [reply tools/cli/admin.go:1289]: how about `listall`? `describe` makes me think that it is a command requiring a key as input and only shows the description of that config.

Comment (mantas-sidlauskas) [reply tools/cli/admin.go]: done :+1:

**Review by Shaddoll — APPROVED** [review]
Stamped, but there is a unit test failure

**Pull request merged**